cmd/internal/find/primary/meta/predicate.go

```diff
@@ -0,0 +1,29 @@
+package meta
+
+import (
+	"github.com/puppetlabs/wash/cmd/internal/find/primary/errz"
+)
+
+// predicate represents a predicate in the meta primary's grammar.
+type predicate func(v interface{}) bool
+
+/*
+Predicate => ObjectPredicate |
+             ArrayPredicate  |
+             PrimitivePredicate
+*/
+func parsePredicate(tokens []string) (predicate, []string, error) {
+	p, tokens, err := try(
+		tokens,
+		parseObjectPredicate,
+		parseArrayPredicate,
+		parsePrimitivePredicate,
+	)
+	if err != nil {
+		if errz.IsMatchError(err) {
+			return nil, nil, errz.NewMatchError("expected either a primitive, object, or array predicate")
```

Comment:
This error message may not be particularly useful. It almost certainly requires reading some docs that don't yet exist.

Reply:
Hmm, it isn't used anywhere b/c `ArrayPredicate` + `ObjectPredicate` only check the match error then return their own error messages. Should I just return `err` here?

Reply:
Sure.